### PR TITLE
[layernorm] hot fix

### PR DESCRIPTION
**Pull request by carlushuang**
need convert to compute-type then do residual add operation

### CODE REVIEW

File: include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_one_pass.hpp [hunk 1]
```diff
@@ -111,8 +111,9 @@ struct Layernorm2dFwdPipelineOnePass
         {
             sweep_tile(x_resi, [&](auto idx) {
                 // compute x = x_resi + x
-                x(idx) = type_convert<YResidualDataType>(x_resi(idx)) +
-                         type_convert<YResidualDataType>(x(idx));
+                auto re_ = type_convert<ComputeDataType>(x_resi(idx)) +
```
Comment (rocking5566) [inline include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_one_pass.hpp:114]: Avoid cast down to fp16 and cast up to fp32.

`auto x = type_convert<ComputeDataType>(x_resi(idx)) + type_convert<ComputeDataType>(x(idx));
store_tile(y_residual_window, cast_tile<XDataType>(x));`

File: include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_two_pass.hpp [hunk 2]
```diff
@@ -170,8 +171,9 @@ struct Layernorm2dFwdPipelineTwoPass
             {
                 sweep_tile(x_resi, [&](auto idx) {
                     // compute x = x_resi + x
-                    x(idx) = type_convert<YResidualDataType>(x_resi(idx)) +
-                             type_convert<YResidualDataType>(x(idx));
+                    auto re_ = type_convert<ComputeDataType>(x_resi(idx)) +
+                               type_convert<ComputeDataType>(x(idx));
+                    x(idx) = type_convert<XDataType>(re_);
```
Comment (rocking5566) [inline include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_two_pass.hpp:176]: Avoid cast down to fp16 and cast up to fp32.

**Review by poyenc — APPROVED** [review]

File: include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_one_pass.hpp (hunk 1 above)
Comment (carlushuang) [reply include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_one_pass.hpp:114]: sure, will create another PR for this~

File: include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_two_pass.hpp (hunk 2 above)
Comment (carlushuang) [reply include/ck_tile/ops/layernorm2d/pipeline/layernorm2d_fwd_pipeline_two_pass.hpp:176]: sure, will create another PR for this~

**Pull request merged**